generate_flocker_certs.sh

```diff
@@ -39,133 +39,256 @@ HELP_MSG="""
+      shift
+      ;;
+    -o=*|--output-dir=*)
+      OUTPUT_DIR="${arg#*=}"
+      shift
+      ;;
+    --force)
+      FORCE_OVERWRITE=true
+      shift
+      ;;
+    -k=*|--key_size=*)
+      KEY_SIZE="${arg#*=}"
+      shift
+      ;;
     -h|--help)
-    echo -e $HELP_MSG && exit 0;
-    shift # past argument with no value
-    ;;
+      echo "${HELP_MSG}" && exit 0;
+      shift
+      ;;
     *)
-            # unknown option
-    ;;
-esac
+      # unknown option
+      ;;
+  esac
 done
-# Config: The user must pass variables based on their cluster setup:
-# or receive defaults
-control_service_dns=${CONTROL_DNS:=""}
+
+control_service_fqdn=${CONTROL_FQDN:=""}
 control_service_ip=${CONTROL_IP:=""}
+key_size=${KEY_SIZE:="4096"}
 
-# fill in the nodes to generate certs for the nodes
-# pass as a list of nodes
-# Example:  --nodes=node1,node2,node3...
-# maybe we want the ability to passa json/yaml/csv to this?
-[ -z "$NODES" ] && echo "Must provide nodes" && exit 1;
-[ -z "${NODES// }" ]  && echo "Must provide at least one node" && exit 1;
-node_count=0
-for n in $NODES
-do
-#echo "$node_count:$n"
-nodes[$node_count]=$n
-((node_count++))
-done
+# Sanity checks
+if [ -z "$NODES" ] || [ -z "${NODES// }" ]; then
+  echo "No nodes provided! Exiting!"
+  echo "${HELP_MSG}"
+  exit 1
+fi
+
+if [ -z "$CLUSTER_NAME" ] || [ -z "${CLUSTER_NAME// }" ]; then
+  echo "No cluster name provided! Exiting!"
+  echo "${HELP_MSG}"
+  exit 1
+fi
 
-# cluster name should be unique for each cluster you have
-cluster_name=${CLUSTER_NAME:="mycluster"}
+# Split nodes into discrete values
+IFS=","
+declare -a nodes
+for node in $NODES; do
+  nodes+=( $node )
+done
+unset IFS
 
-# this is the path to
-openssl_cnf_path=${OPENSSL_FILE:="./openssl.cnf"}
+cluster_name=${CLUSTER_NAME}
+openssl_cnf_path=${OPENSSL_FILE:="openssl.cnf"}
 
-################################################################################
 # set the CERT_HOST_ID environment variable early on since its used in
-# openssl.cnf. Use DNS first if its there.
-if [ $control_service_dns != "" ]; then
-    export CERT_HOST_ID=DNS:control-service,DNS:$control_service_dns
-    control_host=$control_service_dns
-elif [ $control_service_ip != "" ]; then
-    export CERT_HOST_ID=DNS:control-service,IP:$control_service_ip
-    control_host=$control_service_ip
+# openssl.cnf. Use FQDN first if its there.
+if [ "$control_service_fqdn" != "" ]; then
+  export CERT_HOST_ID=DNS:control-service,DNS:$control_service_fqdn
+  control_host=$control_service_fqdn
+elif [ "$control_service_ip" != "" ]; then
+  export CERT_HOST_ID=DNS:control-service,IP:$control_service_ip
+  control_host=$control_service_ip
 else
-    echo "Need Control IP or Control DNS" && exit 1;
+  echo "ERROR! No control service FQDN or IP provided! Exiting!"
+  echo "${HELP_MSG}"
+  exit 1
 fi
 
-# ----------------------------------------
-# generate the CA:
-# ----------------------------------------
+echo "Cleaning up old CA dirs"
+rm -rf ssl
```

Comment:
removing things makes me nervous, especially if someone uses "ssl" outside of this script. Which could be the case if its not `./ssl` Maybe rename this to `./flockerssl` so its obvious.  Same in the openssl conf file.